File: Lecture2-samplecode/Shape.java
// Shape_example.java
// Lecture 02 - Slide Deck
// Shape Class Inheritence Example
// IT2030 - Object Oriented Programming
// (c) SLIIT 2018

class Shape {
    protected String name;
    public Shape() {};
    public Shape (String tname) {
	    name = tname;
    }
    public void print() {
	    System.out.println("\nThen name of shape is: " + name);
    }		 
    public int area(){ return 0;}
}
class Rectangle extends Shape {
       
    protected int width;
    protected int height;
    
    public Rectangle (String tname, int w, int h) {
        super(tname);
	    width = w;
        height = h;
    }

    public int area() {
	   return width * height;
    }

    public void print() {
	    super.print();
	    System.out.println("\nThe height of rectangle is: " + height);
        System.out.println("\nThe width of rectangle is: " + width);
    }
}
class Circle extends Shape {
    protected int radius;
		
    public Circle (String tname, int r) {
        super(tname);
        radius = r;
    }
    public int area() {
	    return (int) (22/7.0 * radius* radius);
    }
    public void print() {
	    super.print();
        System.out.println("\nThe radius of circle is: " + radius);
    }
}

class Main {	
    public static void main(String args[]) {
	    Rectangle R = new Rectangle("Rectangle", 4 , 6);
	    Circle C = new Circle("Circle", 3 );

	    R.print();
	    System.out.println("\nThe area of rectangle is: " + R.area());

	    System.out.println("\n\n*****************************");

	    C.print();
	    System.out.println("\nThe area of circle is: " + C.area());
    }
}

File: Lecture2-samplecode/Shape.cpp
// Shape_example.cpp
// Lecture 02 - Slide Deck
// Shape Class Inheritence Example
// IT2030 - Object Oriented Programming
// (c) SLIIT 2018

# include <iostream>
# include <string>
using namespace std;

class Shape
{
  protected:
	      string name;
  public:
	  Shape();
	  Shape ( string tname) {
	    name = tname;
	  }
	  void print() {
		  cout <<"\nThen name of shape is:"<< name;
	  }		 
	  virtual int area( ){}
};

class Rectangle: public Shape{
    protected:
		  int width;
		  int height;
    public : 
		  Rectangle (string tname, int w, int h) : Shape ( tname){
	    		width = w;
	    		height = h;
        	}
		  int area() {
			  return width * height;
		  }
		  void print() {
			  Shape::print();
			  cout <<"\nThe height of rectangle is: "<< height;
        		cout <<"\nThe width of rectangle is: "<< width;

	     }
};

class Circle: public Shape{
    protected:
		  int radius;
		
    public : 
		  Circle (string tname, int r) : Shape ( tname) {
	    		radius = r;
      }
		  int area( ) {
			  return 2 * 22/7.0 * radius* radius;
		  }
		  void print() {
			  Shape::print();
        cout <<"\nThe radius of circle is: "<< radius;
      }
};


int main() {
	Rectangle R("Rectangle", 4 , 6);
	Circle C("Circle", 3 );

	R.print();
	cout <<"\nThe area of rectangle is: "<<R.area();

	cout<<"\n\n*****************************";

	C.print();
	cout <<"\nThe area of circle is: "<<C.area();
}